libvirt/utils_libvirt.go

```diff
@@ -68,3 +70,70 @@ func getHostArchitecture(virConn *libvirt.Connect) (string, error) {
 
 	return capabilities.Host.CPU.Arch, nil
 }
+
+type GuestCapabilities struct {
+	XMLName xml.Name `xml:"capabilities"`
+	Architectures []struct {
+		Name string `xml:"name,attr"`
+		Wordsize string `xml:"wordsize"`
+		Emulator string `xml:"emulator"`
+		Machines []Machine `xml:"machine"`
+	} `xml:"guest>arch"`
+
+}
+
+type Machine struct {
+	Name string `xml:",chardata"`
+	Canonical string `xml:"canonical,attr"`
+}
```

Comment:
These structs are already defined in libvirt-go-xml: [Maschine](https://github.com/libvirt/libvirt-go-xml/blob/master/capabilities.go#L109-L113) and [CapsGuest](https://github.com/libvirt/libvirt-go-xml/blob/master/capabilities.go#L134-L138).